Document `import_trait_associated_functions` feature (#2312)
rust-lang/reference: src/items/use-declarations.md

* Nameable [items]
* [Enum variants]
* [Built-in types]
* [Associated items] of traits
* [Attributes]
* [Derive macros]
* [`macro_rules`]

r[items.use.path.disallowed]

> They cannot import associated items of structs or enums, [generic parameters], [local variables], paths with [`Self`], or [tool attributes]. More restrictions are described below.

Reviewer: also not from unions, right?

maybe instead 
```suggestion
They cannot import associated items (other than associated items of traits), [generic parameters], [local variables], paths with [`Self`], or [tool attributes]. More restrictions are described below.
```

so that if some other thing with associated items is added later we don't need to update the list here again?

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/reference/2312/afdc77bab886d4455c11247cdd32391bfab636ae..c91676579ea9275a5fc6b85edaed08a7f961866a)*

Author: Good catch. I think we may as well list them out. But the sentence structure gets pretty unwieldy so we could split it into two sentences.

```suggestion
They cannot import associated items of structs, enums or unions. They cannot import [generic parameters], [local variables], paths with [`Self`], or [tool attributes]. More restrictions are described below.
```

Reviewer: What about

```suggestion
They cannot import associated items of non-traits, [generic parameters], [local variables], paths with [`Self`], or [tool attributes]. More restrictions are described below.
```

?

Author: That works but I think it's more straightforward to just be explicit. I don't feel strongly though. Maybe we can get a third opinion.
